src/plugins/mer/meremulatormodeoptionswidget.cpp

```diff
@@ -0,0 +1,230 @@
+/****************************************************************************
+**
+** Copyright (C) 2012 - 2018 Jolla Ltd.
+** Contact: http://jolla.com/
+**
+** This file is part of Qt Creator.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Digia.
+**
+****************************************************************************/
+
+#include "meremulatormodeoptionswidget.h"
+#include "ui_meremulatormodeoptionswidget.h"
+
+#include "mersettings.h"
+
+#include <utils/qtcassert.h>
+
+#include <QMessageBox>
+
+namespace Mer {
+namespace Internal {
+
+MerEmulatorModeOptionsWidget::MerEmulatorModeOptionsWidget(QWidget *parent)
+    : QWidget(parent)
+    , m_ui(new Ui::MerEmulatorModeOptionsWidget)
+    , m_deviceModels(MerSettings::deviceModels())
+{
+    m_ui->setupUi(this);
+    m_ui->optionsWidget->setDeviceModelsNames(m_deviceModels.uniqueKeys());
+
+    connect(m_ui->optionsWidget, &MerEmulatorModeDetailsWidget::screenResolutionChanged,
+            this, &MerEmulatorModeOptionsWidget::deviceModelScreenResolutionChanged);
+    connect(m_ui->optionsWidget, &MerEmulatorModeDetailsWidget::screenSizeChanged,
+            this, &MerEmulatorModeOptionsWidget::deviceModelScreenSizeChanged);
+    connect(m_ui->optionsWidget, &MerEmulatorModeDetailsWidget::optionsChanged,
+            this, &MerEmulatorModeOptionsWidget::deviceModelDconfChanged);
+    connect(m_ui->optionsWidget, &MerEmulatorModeDetailsWidget::deviceModelNameChanged,
+            this, &MerEmulatorModeOptionsWidget::deviceModelNameChanged);
+
+    resetDeviceModelComboBox();
+    m_ui->deviceModelComboBox->setCurrentIndex(0);
+}
+
+MerEmulatorModeOptionsWidget::~MerEmulatorModeOptionsWidget()
+{
+    delete m_ui;
+}
+
+void MerEmulatorModeOptionsWidget::resetDeviceModelComboBox()
+{
+    disconnect(m_ui->deviceModelComboBox, qOverload<int>(&QComboBox::currentIndexChanged),
+            this, &MerEmulatorModeOptionsWidget::updateGui);
+
+    m_ui->deviceModelComboBox->setDeviceModels(m_deviceModels.values());
+
+    connect(m_ui->deviceModelComboBox, qOverload<int>(&QComboBox::currentIndexChanged),
+            this, &MerEmulatorModeOptionsWidget::updateGui);
```

Comment:
I would rather use `QObject::blockSignals` inside `MerDeviceModelComboBox::setDeviceModels()` instead of  temporarily disconnecting the signal here. This would reduce the body of this function to single line, so maybe we could revert it to the older version where you called setDeviceModels directly.

Reply:
Nice one! `QObject::blockSignals` was completely out of my sight. Fixed.